packages/swapper/src/swappers/thorchain/getThorTradeQuote/getTradeQuote.ts

```diff
@@ -71,42 +75,35 @@ export const getThorTradeQuote: GetThorTradeQuote = async ({ deps, input }) => {
       buyAsset.precision,
     )
 
-    const estimatedBuyAssetTradeFeeCryptoHuman = await estimateBuyAssetTradeFeeCrypto(
-      deps,
-      buyAsset,
+    const buyAssetAddressData = await getInboundAddressDataForChain(
+      deps.daemonUrl,
+      buyAsset.assetId,
     )
 
-    const buyAssetUsdRate = await getUsdRate({ deps, input: { assetId: buyAsset.assetId } })
-    const sellAssetUsdRate = await getUsdRate({ deps, input: { assetId: sellAsset.assetId } })
-    const estimatedBuyAssetTradeFeeUsd = bn(buyAssetUsdRate)
-      .times(estimatedBuyAssetTradeFeeCryptoHuman)
-      .toString()
+    const estimatedBuyAssetTradeFeeFeeAssetCryptoHuman = isRune(buyAsset.assetId)
+      ? RUNE_OUTBOUND_TRANSACTION_FEE_CRYPTO_HUMAN.toString()
+      : fromBaseUnit(bnOrZero(buyAssetAddressData?.outbound_fee), THORCHAIN_FIXED_PRECISION)
 
-    const minimumUsdAmount =
-      buyAsset.chainId === KnownChainIds.EthereumMainnet
-        ? MINIMUM_USD_TRADE_AMOUNT_ETHEREUM_NETWORK
-        : MINIMUM_USD_TRADE_AMOUNT
+    const buyAssetChainFeeAssetId = buyAdapter.getFeeAssetId()
 
-    const buyAssetTradeFeeUsdOrMinimum = minimumUsdAmount.gt(estimatedBuyAssetTradeFeeUsd)
-      ? minimumUsdAmount.toString()
-      : estimatedBuyAssetTradeFeeUsd
+    const sellAssetUsdRate = await getUsdRate({ deps, input: { assetId: sellAsset.assetId } })
+    const buyFeeAssetUsdRate = await getUsdRate({
+      deps,
+      input: { assetId: buyAssetChainFeeAssetId },
+    })
 
-    const minimumSellAssetAmountCryptoHuman = (() => {
-      // The 1$ minimum doesn't apply for swaps to RUNE, use OutboundTransactionFee in human value instead
-      if (isRune(buyAsset?.assetId)) return RUNE_OUTBOUND_TRANSACTION_FEE_CRYPTO_HUMAN
+    const buyAssetTradeFeeUsd = bn(buyFeeAssetUsdRate)
+      .times(estimatedBuyAssetTradeFeeFeeAssetCryptoHuman)
+      .toString()
+
+    const minimumSellAssetAmountCryptoHuman = bnOrZero(buyAssetTradeFeeUsd).div(sellAssetUsdRate)
```

Comment:
is it possible sellAssetUsdRate is 0 here?

Reply:
A bad server response could certainly make that possible, which would leave us with the string "Infinity" as the `minimumCryptoHuman` - not ideal.

Guarded against this in https://github.com/shapeshift/lib/pull/1080/commits/4a0205c8e9eba6818f3ae6e14543ffb2611549d7.